And even then the tests don't work against Home Asisstant 2025.12.4.

```pytb
________ ERROR collecting tests/config_flow/test_virtual_power_wled.py _________
ImportError while importing test module '/build/source/tests/config_flow/test_virtual_power_wled.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/3lll9y925zz9393sa59h653xik66srjb-python3-3.13.9/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/config_flow/test_virtual_power_wled.py:21: in <module>
    import custom_components.test.sensor as test_sensor_platform
custom_components/test/sensor.py:15: in <module>
    from homeassistant.const import (
E   ImportError: cannot import name 'POWER_VOLT_AMPERE_REACTIVE' from 'homeassistant.const' (/nix/store/b3ajfxd513xc325xiwy4nx4xhd2br754-homeassistant-2025.12.4/lib/python3.13/site-packages/homeassistant/const.py)
_________________ ERROR collecting tests/strategy/test_wled.py _________________
ImportError while importing test module '/build/source/tests/strategy/test_wled.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/3lll9y925zz9393sa59h653xik66srjb-python3-3.13.9/lib/python3.13/importlib/__init__.py:88: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/strategy/test_wled.py:29: in <module>
    import custom_components.test.sensor as test_sensor_platform
custom_components/test/sensor.py:15: in <module>
    from homeassistant.const import (
E   ImportError: cannot import name 'POWER_VOLT_AMPERE_REACTIVE' from 'homeassistant.const' (/nix/store/b3ajfxd513xc325xiwy4nx4xhd2br754-homeassistant-2025.12.4/lib/python3.13/site-packages/homeassistant/const.py)
```

Should probably be just `VOLT_AMPERE_REACTIVE`.

--- a/pkgs/servers/home-assistant/custom-components/powercalc/package.nix
+++ b/pkgs/servers/home-assistant/custom-components/powercalc/package.nix
@@ -1,0 +1,51 @@
+{
+  lib,
+  buildHomeAssistantComponent,
+  fetchFromGitHub,
+
+  # dependencies
+  numpy,
+
+  # tests
+  home-assistant,
+  pytestCheckHook,
+  pytest-homeassistant-custom-component,
+  pytest-freezegun,
+  aioresponses,
+}:
+
+buildHomeAssistantComponent rec {
+  owner = "bramstroker";
+  domain = "powercalc";
+  version = "1.20.0";
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = "homeassistant-powercalc";
+    tag = "v${version}";
+    hash = "sha256-zncgOnoe24xyVoUBuaUvrL0lqT9COkIPhK3H9SOvd84=";
+  };
+
+  dependencies = [ numpy ];
+
+  nativeCheckInputs = [
+    pytest-homeassistant-custom-component
+    pytestCheckHook
+    aioresponses
+    pytest-freezegun
+  ]
+  ++ home-assistant.getPackages "camera" home-assistant.python.pkgs;
+
+  preCheck = ''
+    patchShebangs --build tests/setup.sh
+    tests/setup.sh
+  '';
+
+  meta = {
+    changelog = "https://github.com/bramstroker/homeassistant-powercalc/releases/tag/${src.tag}";
+    description = "Custom Home Assistant component for virtual power sensors";
+    homepage = "https://github.com/bramstroker/homeassistant-powercalc";
+    maintainers = with lib.maintainers; [ CodedNil ];
+    license = lib.licenses.mit;
+  };
+}

